### PR TITLE
Remove clang warnings from AFQMC configuration

### PR DESCRIPTION
**Pull request by williamfgc**
## Proposed changes

Just cleaning some clang warnings due to unused variables in AFQMC configurations.
Ignore minor clang-formatting changes

## What type(s) of changes does this code introduce?

- Refactoring (no functional changes, no api changes)

### Does this introduce a breaking change?

- Yes

## What systems has this change been tested on?
sulfur

## Checklist

- Yes. This PR is up to date with current the current state of 'develop'
- No. Code added or changed in the PR has been clang-formatted
- No. This PR adds tests to cover any new code, or to catch a bug that is being fixed
- No. Documentation has been added (if appropriate)

### CODE REVIEW

File: src/AFQMC/Hamiltonians/rotateHamiltonian.hpp [hunk 1]
```diff
@@ -476,8 +477,7 @@ inline void rotateHijkl(std::string& type,
       assert(nkcum + K0_ == M_split[nn + nn0]);
       if (M_split[nn + nn0 + 1] == M_split[nn + nn0])
         continue;
-      int nblk       = Qknum[nn];
-      long ntermscum = 0;
```
Comment (ye-luo) [inline src/AFQMC/Hamiltonians/rotateHamiltonian.hpp:480]: This was used in an assertion. Thus breaks Debug builds. Unfortunately, current CI Debug doesn't cover AFQMC.

Comment (williamfgc) [reply src/AFQMC/Hamiltonians/rotateHamiltonian.hpp:480]: Good catch, would using `#ifndef NDEBUG` work? I see it's used in other parts of the code.

Comment (ye-luo) [reply src/AFQMC/Hamiltonians/rotateHamiltonian.hpp:480]: `#ifndef NDEBUG` is likely a solution to keep the check for debug build without printing a warning in the release builds.

Comment (prckent) [reply src/AFQMC/Hamiltonians/rotateHamiltonian.hpp:480]: Works for me also. Great to see the warning reductions.

**Review by prckent — APPROVED** [review]
LGTM

Comments have been addressed and it looks like the needed guards are there now.

In the interests of making progress on the PR backlog, will approve and merge. Other reviewers are welcome to block if they wish to check the build by hand.

**Pull request merged**